### PR TITLE
Add tllist

### PR DESCRIPTION
**Pull request by danimtb**
From #8588 and the changes proposed in the reviews of such PR (including https://github.com/aacebedo/conan-center-index/pull/1)

closes #8588

### CODE REVIEW

**Review by SSE4 — DISMISSED** [review]

File: recipes/tllist/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import os
+
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.43.0"
```
Comment (SpaceIm) [inline recipes/tllist/all/conanfile.py:7]:
```suggestion
required_conan_version = ">=1.36.0"
```

File: recipes/tllist/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                  destination=self._source_subfolder, strip_root=True)
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        self.copy("*.h", src=self._source_subfolder, dst="include")
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
```
Comment (SpaceIm) [inline recipes/tllist/all/conanfile.py:39]:
```suggestion
    def package_info(self):
        self.cpp_info.set_property("pkg_config_name", "tllist")
```

Comment (SpaceIm) [reply recipes/tllist/all/conanfile.py:39]: https://codeberg.org/dnkl/tllist/src/tag/1.0.5/meson.build#L11-L16

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/tllist/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    settings = "os", "arch", "build_type", "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("Only Linux supported")
```
Comment (danimtb) [inline recipes/tllist/all/conanfile.py:26]:
```suggestion
```

File: recipes/tllist/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        self.copy("*.h", src=self._source_subfolder, dst="include")
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.set_property("pkg_config_name", "tllist")
+        self.cpp_info.includedirs.append(os.path.join("include"))
```
Comment (danimtb) [inline recipes/tllist/all/conanfile.py:37]: "include" is the default value, so this shouldn't be needed
```suggestion
        self.cpp_info.includedirs.append(os.path.join("include"))
```

File: recipes/tllist/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import os
+
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
```
Comment (SpaceIm) [inline recipes/tllist/all/conanfile.py:4]:
```suggestion
from conans import ConanFile, tools
```

**Review by SpaceIm — APPROVED** [review]

File: recipes/tllist/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    homepage = "https://codeberg.org/dnkl/tllist"
+    description = "A C header file only implementation of a typed linked list."
+    topics = ("list", "utils", "typed-linked-list")
+    settings = "os", "arch", "build_type", "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
```
Comment (danimtb) [inline recipes/tllist/all/conanfile.py:20]: No idea why it fails on windows and I don't have the time to figure it out. It fails on the test_package side. As the original PR was restricting build different from Linux, I am ok restricting them now only for windows and adding a proper fix later.
```suggestion
    def validate(self):
        # FIXME: Fails in test_pacakge with the Windows configuration
        if self.settings.os == "Windows":
            raise ConanInvalidConfiguration("Windows not supported")

```

File: recipes/tllist/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    homepage = "https://codeberg.org/dnkl/tllist"
+    description = "A C header file only implementation of a typed linked list."
+    topics = ("list", "utils", "typed-linked-list")
+    settings = "os", "arch", "build_type", "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+    def validate(self):
```
Comment (danimtb) [inline recipes/tllist/all/conanfile.py:20]:
```suggestion
        return "source_subfolder"

    def validate(self):
```

File: recipes/tllist/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        # FIXME: Fails in test_pacakge with the Windows configuration
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows not supported")
```
Comment (SpaceIm) [inline recipes/tllist/all/conanfile.py:24]: only Visual Studio

**Review by uilianries — APPROVED** [review]

File: recipes/tllist/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
+from conan import ConanFile
```
Comment (SpaceIm) [inline recipes/tllist/all/conanfile.py:1]: what is the requirement in terms of conan version?

File: recipes/tllist/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+from conans import tools
+from conans.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.36.0"
```
Comment (danimtb) [inline recipes/tllist/all/conanfile.py:6]:
```suggestion
required_conan_version = ">=1.43.0"
```

File: recipes/tllist/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    topics = ("list", "utils", "typed-linked-list")
+    settings = "os", "arch", "build_type", "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        # FIXME: Fails in test_pacakge with the Visual Studio compiler
```
Comment (SpaceIm) [inline recipes/tllist/all/conanfile.py:24]:
```suggestion
        # tllist relies on __typeof__, not implemented in Visual Studio
```

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**